eng/tools/suppressions/test/e2e.test.ts

```diff
@@ -0,0 +1,38 @@
+import { expect, test } from "vitest";
+import { getSuppressions } from "../src/suppressions.js";
+import { join } from "path";
+
+async function getTestSuppressions(path: string, tool: string = "TestTool") {
+  return await getSuppressions(tool, join(__dirname, "e2e", path));
```

Comment:
General design remark:

While `getTestSuppressions` works for now, I figured out that what works for me very well for organizing my "test infra logic" is to introduce a `-Fixture` pattern. So each class/type `Foo` ends up having a trio (in C#): `Foo`, `FooTests` and `FooFixture`. `Foo` is the system under test. `FooTests` is the test suite. And `FooFixture` is the intermediate layer of "test infra" that provides all the test assets required by `FooTests`.

This is a recursive pattern. That is: bigger fixtures are constructed by using constituent fixtures. So if you imagine having an `Employee` class and `Organization` composed of employees, then `OrganizationFixture` can compose its test inputs using `EmployeeFixture`s.

For examaple, if you want to test some complex organization hierarchy, the `OrganizationTests` 
test looks like that:

(in C# like pseudocode)

``` csharp
class OrganizationTests
{
  public TestComplexOrg()
  {
    var orgFixture = new OrganizatonFixture(); 
    assert(orgFixture.complexOrg1).isEqualTo(...);
  }
}

class OrganizationFixture
{
  public OrganizationFixture()
  {
     var employeeFixture = new EmployeeFixture();
     Organization complexOrg1 = OrganizationFixture.From({ employees: [ employeeFixture.employee1, employeeFixture.employee2, ... })
  }
}

Reply:
I'm fine if you want to try this in specs-model.  My initial reaction, it seems a bit overkill, but it will be easier to comment on specific code.